### PR TITLE
Enhanced testing

### PR DESCRIPTION
**Pull request by zaherg**
This is an opinionated enhancement for the testing, instead of using docker we pull wordpress test-libs and use it for out feature testing.

The tests are still the same old one, except a few changes, so they are all accredited to the [original creator](https://github.com/rhubarbgroup/redis-cache/pull/329) but due to the changes I made, the ownership has changed.

We can use this as a base to build upon and add more tests.

> **Warning** This is an opinionated enhancement
>
> You should run `./setup-test.sh` to setup the test environment locally.
> If you want to change any of they the test environment variable, create a new file called `config.sh` and add the following to it:
> 
> ```bash
> GIT_BRANCH=trunk # the wordpress version you want to test against
> DB_HOST=127.0.0.1
> DB_PASS=mysuperpassword # your password, default empty
> DB_NAME=yourdatabase # your database name, default `wp_tests`
> DB_USER=youruser # your database user, default `root`
> ```
> 
> Creating `config.sh` file is not mandatory, as long as you are using the default values.

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+// function get_file_data($file, $header)
+// {
+//     return ['Version' => '2.1.0'];
+// }
+
+// function register_activation_hook(string $file, callable $callback)
+// {
+//     return true;
+// }
```
Comment (tillkruss) [inline tests/bootstrap.php:65]: Can we remove these?

File: tests/TestCase.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        // Your own additional tear down.
+        parent::tear_down();
+    }
+
+    protected function init_cache()
+    {
+        global $wp_object_cache;
+        $cache_class = get_class($wp_object_cache);
+        $cache = new $cache_class();
+        $cache->add_global_groups(['global-cache-test', 'users', 'userlogins', 'usermeta', 'user_meta', 'useremail', 'userslugs', 'site-transient', 'site-options', 'blog-lookup', 'blog-details', 'rss', 'global-posts', 'blog-id-cache', 'networks', 'sites', 'site-details']);
```
Comment (tillkruss) [inline tests/TestCase.php:34]: Let's use new lines here.

File: tests/ExampleTest.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Unit;
+
+use Tests\TestCase;
+
+/**
+ * @internal
+ * @coversNothing
+ */
+class ExampleTest extends TestCase
```
Comment (tillkruss) [inline tests/ExampleTest.php:13]: Should we drop this file?

File: setup-test.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#! /usr/bin/env sh
```
Comment (tillkruss) [inline setup-test.sh:1]: Let's put this file in `bin`.

File: README.md [hunk 5]
```diff
@@ -48,4 +48,24 @@ To see a list of all available WP-CLI commands, please see the [WP CLI commands
 
 ### Development
 
-Head over to the [Docker Development wiki page](https://github.com/rhubarbgroup/redis-cache/wiki/Docker-Development) to spin up various Redis setups.
```
Comment (tillkruss) [inline README.md:51]: We should move this to the Wiki page. Do you have permission to update it?

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (tillkruss) [reply README.md:51]: Sure 👍

File: tests/TestCase.php (hunk 2 above)
Comment (tillkruss) [reply tests/TestCase.php:35]: I meant the array 🙃

Comment (zaherg) [reply tests/TestCase.php]: the new line after `global` add more readability to it, so win/win

**Pull request merged**